python/lsst/pipe/base/pipelineTask.py

```diff
@@ -244,22 +244,77 @@ def getDatasetTypes(cls, config, configClass):
                 dsTypes[key] = cls.makeDatasetType(value)
         return dsTypes
 
-    def run(self, *args, **kwargs):
+    def runWithOutputs(self, inputData, outputDataIds):
```

Comment:
I have very strong reservations about a method called runWithOutputs that in the default implementation does nothing with the outputDataIds variable, or anything output related